dgc/GreenCertificateDecoder.cs

```diff
@@ -0,0 +1,54 @@
+using ICSharpCode.SharpZipLib.Zip.Compression.Streams;
+using NL.MinVWS.Encoding;
+using System;
+using System.IO;
+
+namespace DGC
+{
+    public class GreenCertificateDecoder
+    {
+        /// <summary>
+        /// Decodes base45 encoded string -> Deflate -> COSE -> CBOR -> arbitrary Json String
+        /// </summary>
+        /// <param name="base45String">Base45 Encoded string</param>
+        /// <returns>Cose object and a digital green card v1 object</returns>
+        public CWT Decode(string base45String)
+        {
+            if (!base45String.StartsWith("HC1:"))
+                throw new ArgumentException("Base45 string not valid according to specification");
+
+            base45String = base45String.Substring(4);
+            var decodedBytes = Base45Encoding.Decode(base45String);
+
+            var coseBytes = DeflateToCoseBytes(decodedBytes);
+
+            var coseOBj = Sign1CoseMessage.DecodeFromBytes(coseBytes);
+
+            CWT cwt = CWT.DecodeFromBytes(coseOBj);
+            
+            return cwt;
+        }
+
+        private byte[] DeflateToCoseBytes(byte[] decodedBytes)
+        {
+            if (decodedBytes[0] == 0x78 && decodedBytes[1] == 0xDA)
```

Comment:
Nice, I definitely prefer this. However, spec says this zlib compression step is mandatory.
I'm more than happy to leave this as-is since it will work in any case and shows that it is essentially a very trivial matter to detect whether compression is present at runtime or not